operator/pkg/controllers/etcd/pod.go

```diff
@@ -187,10 +169,23 @@ func podSpecFor(controlPlane *v1alpha1.ControlPlane) *v1.PodSpec {
 	}
 }
 
+func generateEtcdBootstrapCmd(controlPlane *v1alpha1.ControlPlane) string {
+	return fmt.Sprintf(etcdBootstrapCmd, etcdctlEndpoints(controlPlane), advertizePeerURL(controlPlane),
+		initialClusterFlag(controlPlane), advertizeClusterURL(controlPlane))
+}
+
 func initialClusterFlag(controlPlane *v1alpha1.ControlPlane) string {
+	return endPointFlagGenerator(controlPlane, "%[1]s-etcd-%[2]d=https://%[1]s-etcd-%[2]d.%[1]s-etcd.%[3]s.svc.cluster.local:2380")
+}
+
+func etcdctlEndpoints(controlPlane *v1alpha1.ControlPlane) string {
+	return endPointFlagGenerator(controlPlane, "https://%[1]s-etcd-%[2]d.%[1]s-etcd.%[3]s.svc.cluster.local:2379")
+}
+
+func endPointFlagGenerator(controlPlane *v1alpha1.ControlPlane, format string) string {
 	nodes := make([]string, 0)
 	for i := 0; i < controlPlane.Spec.Etcd.Replicas; i++ {
```

Comment:
I'm not sure abstracting this for loop helps readability. It took me a while to figure out how this was wired up. I think better naming could help (e.g. [curry](https://en.wikipedia.org/wiki/Currying), but it may just be simpler to unroll the for loop and repeat yourself, since it's only used in two locations.